application_sandbox/stencil/mirror.vert

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#version 450
+#include "models/model_setup.glsl"
+
+layout (location = 1) out vec2 texcoord;
+
+layout (binding = 0, set = 0) uniform camera_data {
+    layout(column_major) mat4x4 projection;
+};
+
+layout (binding = 1, set = 0) uniform model_data {
+    layout(column_major) mat4x4 transform;
+};
+
+void main() {
+    mat4x4 translation = {
```

Comment:
Instead of doing it this way, you could have alternatively used the same shader as the cube, but passed in a different model_data.transform that pre-multiplies these matrices. (It would likely be more efficient as well, since you are doing 2 extra matrix multiplies / vertex).

That being said, this is drawing a cube, so efficiency is not really our #1 concern.